tests/unit/cluster/slot-migration.tcl

```diff
@@ -0,0 +1,338 @@
+proc get_open_slots {srv_idx} {
+  foreach line [split [R $srv_idx cluster nodes] "\n"] {
+    set line [string trim $line]
+    if {$line eq {}} continue
+    if {[regexp {myself} $line] == 0} continue
+    set slots {}
+    regexp {\[.*} $line slots
+    return $slots
+  }
+}
+
+proc get_cluster_role {srv_idx} {
+  foreach line [split [R $srv_idx cluster nodes] "\n"] {
+    set line [string trim $line]
+    if {$line eq {}} continue
+    if {[regexp {myself} $line] == 0} continue
+    set role {}
+    regexp {myself,(\w+)} $line -> role
+    return $role
+  }
+}
+
+proc wait_for_role {srv_idx role} {
+  set node_timeout [lindex [R 0 config get cluster-node-timeout] 1]
+  # wait for a gossip cycle for states to be propagated throughout the cluster
+  after $node_timeout
+  wait_for_condition 100 100 {
+    [lindex [split [R $srv_idx role] " "] 0] eq $role
+  } else {
+    fail "R $srv_idx didn't assume the replication $role in time"
+  }
+  wait_for_condition 100 100 {
+      [get_cluster_role $srv_idx] eq $role
+  } else {
+      fail "R $srv_idx didn't assume the cluster $role in time"
+  }
+}
+
+proc wait_for_slot_state {srv_idx pattern} {
+  wait_for_condition 100 100 {
+    [get_open_slots $srv_idx] eq $pattern
+  } else {
+    fail "incorrect slot state on R $srv_idx"
+  }
+}
+
+start_cluster 3 3 {tags {external:skip cluster}
+                   overrides {cluster-allow-replica-migration no
+                              cluster-node-timeout 1000} } {
+
+set node_timeout [lindex [R 0 config get cluster-node-timeout] 1]
```

Comment:
Why not do proper indentation? The only reason it's not always done in the tests (afaik) is to avoid touching the git history of a lot of old code, but this is a new file.

Reply:
I thought that is the best formatting :). Jokes aside, happy to make any formatting change if someone can point me to the right template.

Reply:
Well, I thought indent the body of { ... } is the normal, as is done in all the other { ... }. Not that it matters much though, if you prefer to avoid it.

Reply:
reformatted the test code so start_server is not longer multi-lines and fix some indentation issues. I think it looks like the majority of the test code but please let me know if there are still things missed.

Reply:
I see. I've fixed the indentation now. Also added a new test.